extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -508,24 +608,62 @@ public Domain getDomain( final String domainId ) {
   @Override
   public Set<String> getDomainIds() {
     logger.debug( "getDomainIds()" );
+    return getDomainIdsHelper( () -> metadataMapping.getDomainIds() );
+  }
+
+  /**
+   * Return a list of all the domain ids in the repository of the data source type Metadata.
+   * See {@link #getDomainIds()} for similar functionality.
+   *
+   * @return the metadata domain Ids.
+   */
+  @Override
+  public Set<String> getMetadataDomainIds() {
+    logger.debug( "getDomainIdsMetadata()" );
+    return getDomainIdsHelper( () -> dataSourceTypeMapping.getDatasourceType(
+            PentahoDatasourceType.Metadata.toString() ) );
+  }
+
+  /**
+   * Return a list of all the domain ids in the repository of the data source type Metadata.
+   * See {@link #getDomainIds()} for similar functionality.
+   *
+   * @return the data source wizard domain Ids.
+   */
+  @Override
+  public Set<String> getDataSourceWizardDomainIds() {
```

Comment:
[#1 Add function to get only Metadata Domains Ids and add functions to get only DataSourceWizards Domains Ids](https://jira.pentaho.com/browse/BACKLOG-36127)